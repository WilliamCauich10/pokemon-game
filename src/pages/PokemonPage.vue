<template>
  <h1 v-if="!pokemon">Espere por favor...</h1>

  <div v-else>
    <h1>¿Quién es este pokémon?</h1>
    <!-- TODO img -->
    <PokemonPicture :pokemonId="pokemon.id" :showPokemon="showPokemon" />
    <!-- TODO Opciones @selection ="checkAnswer(1, $event)" //el 1 es el numero de elementos-->
    <PokemonOptions
      :pokemons="pokemonArr" 
      @selection ="checkAnswer"
      />

    <!-- template para no poner div y modificar el css -->
      <template v-if="showAnswer">
        <h2 class="fade-in" > {{ message }} </h2>
        <button @click="newGame">
          Nuevo Juego
        </button>
      </template>

  </div>

</template>

<script>
import PokemonOptions from '@/components/PokemonOptions'
import PokemonPicture from '@/components/PokemonPicture'

import getPokemonOptions from "@/helpers/getPokemonOptions";

// console.log( getPokemonOptions());

export default {
    name: 'PokemonPage',
    components:{
        PokemonPicture,
        PokemonOptions
    },
    data() {
      return {
        pokemonArr: [],
        pokemon: null,
        showPokemon: false,
        showAnswer: false,
        message: ''
      }
    },
    methods: {
      async mixPokemonArray() {
        this.pokemonArr = await getPokemonOptions()

        const rndInt = Math.floor( Math.random() * 4 )
        this.pokemon = this.pokemonArr[ rndInt ]
      },
      checkAnswer( selectedId) {
        // this.showPokemon = selectedId ===  this.pokemon.id ? true : false
        this.showPokemon = true
        // console.log('pokemon page called', selectedId, this.pokemon.id);
        this.showAnswer = true
        this.message = selectedId === this.pokemon.id ? `Correcto, ${ this.pokemon.name }` : `Oops, era ${ this.pokemon.name } `

      },
      newGame() {
        this.showPokemon = false
        this.showAnswer = false
        this.pokemonArr = []
        this.pokemon = null
        this.mixPokemonArray()
      }
    },
    mounted() {
      this.mixPokemonArray()
    }
    
}
</script>
