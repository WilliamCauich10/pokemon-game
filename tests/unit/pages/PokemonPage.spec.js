import { shallowMount, mount } from "@vue/test-utils";
import PokemonPage from '@/pages/PokemonPage'
import { pokemons } from "../mocks/pokemons.moc";

describe('PokemonPage Component', () => { 

    let wrapper

    beforeEach( () => { 
        wrapper = shallowMount( PokemonPage ) 
    })
  
    test('debe de hacer match con el snapshot', () => { 
         expect( wrapper.html() ).toMatchSnapshot( )
    })

    test('debe de llamar mixPokemonArray al montar', () => { 
      
        const mixPokemonArraySpy = jest.spyOn( PokemonPage.methods, 'mixPokemonArray' )
        const wrapper = shallowMount( PokemonPage ) 

        expect( mixPokemonArraySpy ).toHaveBeenCalled(  )
    })

    test('debe de hacer match con el snapshot cuando cargan los pokemons', () => { 
         //con mount crea el ciclo de vida del componente, el shallowMount no 
        const wrapper = shallowMount( PokemonPage, {
            data() {
                return {
                    pokemonArr: pokemons,
                    pokemon: pokemons[0],
                    showPokemon: false,
                    showAnswer: false,
                    message: ''
                }
            }
        })

        expect( wrapper.html() ).toMatchSnapshot( )
    })
})