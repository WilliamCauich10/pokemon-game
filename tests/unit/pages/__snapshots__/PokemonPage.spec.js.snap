// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonPage Component debe de hacer match con el snapshot 1`] = `<h1>Espere por favor...</h1>`;

exports[`PokemonPage Component debe de hacer match con el snapshot cuando cargan los pokemons 1`] = `
<div>
  <h1>¿Quién es este pokémon?</h1><!-- TODO img -->
  <pokemon-picture-stub pokemonid="1" showpokemon="false"></pokemon-picture-stub><!-- TODO Opciones @selection ="checkAnswer(1, $event)" //el 1 es el numero de elementos-->
  <pokemon-options-stub pokemons="[object Object],[object Object],[object Object],[object Object]"></pokemon-options-stub><!-- template para no poner div y modificar el css -->
  <!--v-if-->
</div>
`;
